### DISCUSSION TITLE
When using a substream, at what point does Frigate start using the main stream?

### DISCUSSION THREAD

**Discussion by leccelecce** [blakeblackshear/frigate#5328] [Ask A Question] [answered]
I can't quite figure this out from the docs, so would like to send a PR to add a few sentences of clarification. Let's say I have a 4k main stream and a 720p substream, configured for record and detect roles respectively. I'm trying to validate which ones get used where.

Motion detection: [detect]
Snapshot in UI: [record]
Snapshot via MQTT: [detect] ?
Object inference: ?
Recording: [record]

**Comment by blakeblackshear** [comment, marked as answer]
Anything that requires decoding or analysis of the image uses the detect stream.

```
Motion detection: [detect]
Snapshot in UI: [detect]
Snapshot via MQTT: [detect]
Object inference: [detect]
Recording: [record]
```

**Comment by leccelecce** [comment]
Argh! I thought this was complicated because I downloaded a snapshot image via the UI and it was 3840x2160...but then realised that I had briefly set the detect role to use the main stream for testing. Ignore me!
